### PR TITLE
Most of Tumble

### PR DESCRIPTION
**Pull request by SvenLindstrom**
Still TODO:
Bookmark and search screen are pretty much finished.
Bookmark details sheet is missing color picker.
User Resources are still work in progress.
Settings screen is missing external connections.
Nav deep links are not setup.
SettingsViewModel is missing function implementation.
Bookmark Settings unable to delete schedule.
Notification implementation.

### CODE REVIEW

File: app/src/main/java/tumble/app/tumble/data/api/auth/AuthApiService.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import retrofit2.http.GET
+import retrofit2.http.Header
+import retrofit2.http.POST
+import retrofit2.http.Query
+import tumble.app.tumble.domain.models.network.NetworkRequest
+import tumble.app.tumble.domain.models.network.NetworkResponse
+
+interface AuthApiService {
+    @POST("/api/users/login")
+    suspend fun loginUser(
```
Comment (adisve) [inline app/src/main/java/tumble/app/tumble/data/api/auth/AuthApiService.kt:14]: When debugging/running the app, signing in does not work for a known user. Is this intended for now?

File: app/src/main/java/tumble/app/tumble/presentation/viewmodels/SearchViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    var selectedSchool = mutableStateOf<School?>(null)
+    var universityImage by mutableStateOf<Int?>(null)
+    var searching = mutableStateOf(false)
+    var searchBarText = mutableStateOf("")
+
+    private var currentSearchJob: Job? = null
+
+    val schools: List<School> by lazy { schoolManager.getSchools() }
+
+    fun search(query: MutableState<String>, selectedSchoolId: Int){
```
Comment (adisve) [inline app/src/main/java/tumble/app/tumble/presentation/viewmodels/SearchViewModel.kt:46]: When searching for a known schedule, the tab loads and then goes back to default uni selection. Is this intended for now?

File: app/src/main/java/tumble/app/tumble/presentation/viewmodels/SettingsViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    fun removeNotifications(id: String, events: List<Event>){
+        //TODO
+    }
+
+    fun clearAllNotifications(){
+        //TODO
+    }
+
+    fun rescheduleNotifications(previousOffset: Int, newOffset: Int){
```
Comment (adisve) [inline app/src/main/java/tumble/app/tumble/presentation/viewmodels/SettingsViewModel.kt:99]: When changing the notification offset, the app crashes and can not be started again. Is this intended for now?

**Review by adisve — CHANGES_REQUESTED** [review]
Incredibly nice work and very good code. I like what you have done! There are some minor issues that I commented on that you could check out before I accept it.

File: app/src/main/java/tumble/app/tumble/presentation/viewmodels/SearchViewModel.kt (hunk 2 above)
Comment (SvenLindstrom) [reply app/src/main/java/tumble/app/tumble/presentation/viewmodels/SearchViewModel.kt:46]: This works when connected to the server running locally

File: app/src/main/java/tumble/app/tumble/presentation/viewmodels/SettingsViewModel.kt (hunk 3 above)
Comment (SvenLindstrom) [reply app/src/main/java/tumble/app/tumble/presentation/viewmodels/SettingsViewModel.kt]: This has been fixed

**Review by adisve — APPROVED** [review]

**Comment by adisve** [comment]
You are welcome to merge it

**Pull request merged**